src/application.ts

```diff
@@ -51,14 +49,36 @@ export default class Application {
     op: Operation,
     processor: OperationProcessor<Resource>
   ): Promise<OperationResponse> {
-    const result = await processor.execute(op);
+    const result = await processor.execute(await this.deserializeResource(op));
     return this.buildOperationResponse(result);
   }
 
   async createTransaction(ops: Promise<OperationResponse>[]) {
     return await Promise.all(ops);
   }
 
+  async deserializeResource(op: Operation) {
+    const resourceClass = await this.resourceFor(op.ref.type);
+    const schemaRelationships = resourceClass.schema.relationships;
+    if (op.data && op.data.attributes) {
```

Comment:
Something I usually do for easier readibility: If a condition is relevant to a large portion of a function's code, you can negate it and do an earlier exit:

```ts
if (!op.data || !op.data.attributes) {
  return op;
}

// Code that works with the passing condition.
// yadayadayada...

return op;
```

Yes, it has the "downside" of a _double return_. I'll leave it up to you to decide if this works here or not.

Reply:
You're right, I'll fix it right away